internal/impl/cypher/output_cypher_test.go

```diff
@@ -0,0 +1,259 @@
+	_ = resource.Expire(900)
+
+	return neo4jDockerAddress, nil
+}
+
+func createCypherOutputFromYaml(template string) (s output.Streamed, err error) {
+
+	conf, err := testutil.OutputFromYAML(template)
+	if err != nil {
+		return nil, err
+	}
+
+	s, err = mock.NewManager().NewOutput(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func sendMessages(t *testing.T, s output.Streamed) {
+	sendChan := make(chan message.Transaction)
+	resChan := make(chan error)
+	if err := s.Consume(sendChan); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Cleanup(func() {
+		s.TriggerCloseNow()
+
+		ctx, done := context.WithTimeout(context.Background(), time.Second*10)
+		assert.NoError(t, s.WaitForClose(ctx))
+		done()
+	})
+
+	inputs := []string{
+		`{"name":"Alice"}`, `{"name":"Bob"}`, `{"name":"Carol"}`, `{"name":"Dan"}`,
+	}
+
+	for _, input := range inputs {
+		testMsg := message.QuickBatch([][]byte{[]byte(input)})
+		select {
+		case sendChan <- message.NewTransaction(testMsg, resChan):
+		case <-time.After(time.Second * 60):
+			t.Fatal("Action timed out")
+		}
+
+		select {
+		case res := <-resChan:
+			if res != nil {
+				t.Fatal(res)
+			}
+		case <-time.After(time.Second * 60):
+			t.Fatal("Action timed out")
+		}
+	}
+}
+
+func checkNeo4j(localURL string, basicAuth bool) (listOfNodeNames []string) {
+
+	ctx := context.Background()
+
+	var driver neo4j.DriverWithContext
+	if basicAuth {
+		driver, _ = neo4j.NewDriverWithContext(localURL, neo4j.BasicAuth("neo4j", "sparkling_brazilian_orange_456", ""))
+	} else {
+		driver, _ = neo4j.NewDriverWithContext(localURL, neo4j.NoAuth())
+	}
+
+	results, _ := neo4j.ExecuteQuery(ctx, driver,
+		"match (n) return n",
+		map[string]any{},
+		neo4j.EagerResultTransformer,
+		neo4j.ExecuteQueryWithDatabase("neo4j"),
+	)
+
+	var listOfNamesFromDB []string
+
+	for _, record := range results.Records {
+		dictionary := record.AsMap()
+		x := dictionary["n"].(dbtype.Node)
+		listOfNamesFromDB = append(listOfNamesFromDB, x.Props["name"].(string))
+	}
+
+	return listOfNamesFromDB
+}
+
+func TestIntegrationCypherOutput(t *testing.T) {
+	integration.CheckSkip(t)
+	t.Parallel()
+
+	t.Run("cypher_output_no_auth", func(t *testing.T) {
+		localURL, err := setupNeo4j([]string{"NEO4J_AUTH=none"})
+		require.NoError(t, err)
+		testCypherOutputNoAuth(t, localURL)
+	})
+
+	t.Run("cypher_output_basic_auth", func(t *testing.T) {
+		localURL, err := setupNeo4j([]string{"NEO4J_AUTH=neo4j/sparkling_brazilian_orange_456"})
+		require.NoError(t, err)
+		testCypherOutputBasicAuth(t, localURL)
+	})
+}
+
+func testCypherOutputNoAuth(t *testing.T, localURL string) {
+	template := fmt.Sprintf(`
+cypher:
+  database: neo4j
+  uri: %s
+  no_auth: true
+  query: |
+    CREATE (p:Person {name: $name}) RETURN p
+  values:
+    name: ${! json("name") }
+  batching:
+    count: 0
+  `, localURL)
+
+	s, err := createCypherOutputFromYaml(template)
+	require.NoError(t, err)
+
+	// send the messages
+	sendMessages(&testing.T{}, s)
```

Comment:
You need to pass the `testing.T` fixture to these sub-tests, otherwise we'll have issues with test context
```suggestion
	sendMessages(t, s)
```